std/typecons.d

```diff
@@ -143,25 +146,39 @@ public:
         u._p = null;
     }
 
-    ~this()
+    // whether a RefT instance can be @safely destroyed at end of scope
+    static if (__traits(compiles, (RefT t) @safe {})) @trusted ~this()
     {
-        debug(Unique) writeln("Unique destructor of ", (_p is null)? null: _p);
-        if (_p !is null) destroy(_p);
-        _p = null;
+        debug(Unique) writeln("Unique @trusted destructor of ", (_p is null)? null: _p);
+        if (_p !is null)
+        {
+            destroy(_p);
+            _p = null;
+        }
     }
+    else this()
+    {
+        debug(Unique) writeln("Unique @system destructor of ", (_p is null)? null: _p);
+        if (_p !is null)
+        {
+            destroy(_p);
+            _p = null;
+        }
+    }
+
     /** Returns whether the resource exists. */
     @property bool isEmpty() const
     {
         return _p is null;
     }
-    /** Transfer ownership to a $(D Unique) rvalue. Nullifies the current contents. */
+    /** Transfer ownership to a $(D Unique) rvalue. Nullifies the current contents.
+    Same as calling std.algorithm.move on it.
+    */
     Unique release()
     {
-        debug(Unique) writeln("Release");
-        auto u = Unique(_p);
-        assert(_p is null);
-        debug(Unique) writeln("return from Release");
-        return u;
+        debug(Unique) writeln("Unique Release");
+        import std.algorithm : move;
```

Comment:
Be more specific with the imports as then the compiler might not need to open and parse the `std.algorithm` package, but only the subpackage (that's why CircleCi fails), here it's `std.algorithm.mutation`